pyethapp/jsonrpc.py

```diff
@@ -928,9 +930,11 @@ def getBlockByNumber(self, block_id, include_transactions):
     def getTransactionByHash(self, tx_hash):
         try:
             tx, block, index = self.chain.chain.index.get_transaction(tx_hash)
+            if self.chain.chain.in_main_branch(block):
```

Comment:
Is that check really necessary? Does it solve any issues with ´rpc-tests´? It prevents us from accessing a branched block's transactions (by hash) for any debugging purposes, right?